backend/src/north/north-rest/north-rest.ts

```diff
@@ -0,0 +1,259 @@
+import NorthConnectorRepository from '../../repository/config/north-connector.repository';
+import ScanModeRepository from '../../repository/config/scan-mode.repository';
+import { BaseFolders } from '../../model/types';
+import { filesExists } from '../../service/utils';
+import FormData from 'form-data';
+import { request, ProxyAgent } from 'undici';
+import { URL } from 'node:url';
+
+/**
+ * Class Console - display values and file path into the console
+ */
+export default class NorthREST extends NorthConnector<NorthRESTSettings> {
+  constructor(
+    configuration: NorthConnectorEntity<NorthRESTSettings>,
+    encryptionService: EncryptionService,
+    northConnectorRepository: NorthConnectorRepository,
+    scanModeRepository: ScanModeRepository,
+    logger: pino.Logger,
+    baseFolders: BaseFolders
+  ) {
+    super(configuration, encryptionService, northConnectorRepository, scanModeRepository, logger, baseFolders);
+  }
+
+  async handleContent(data: OIBusContent): Promise<void> {
+    switch (data.type) {
+      case 'raw':
+        return this.handleFile(data.filePath);
+      case 'time-values':
+        throw new Error('Can not manage time values');
+    }
+  }
+
+  /**
+   * Handle the file by sending it over to the specified endpoint
+   */
+  async handleFile(filePath: string): Promise<void> {
+    if (!(await filesExists(filePath))) {
+      throw new Error(`File ${filePath} does not exist`);
+    }
+
+    const endpoint = this.connector.settings.endpoint;
+
+    // Get query params
+    const queryParams = this.getQueryParams();
+
+    // Create FormData with file
+    const { base } = path.parse(filePath);
+    const form = new FormData();
+    const fileStream = createReadStream(path.resolve(filePath));
+    form.append('file', fileStream, { filename: base });
+
+    const headers: Record<string, string> = form.getHeaders();
+
+    // Get proxy agent if needed
+    const proxyAgent = await this.getProxyAgent();
+
+    // Add authorization header if available
+    const authHeader = await this.getAuthorizationHeader();
+    if (authHeader) {
+      headers['Authorization'] = authHeader;
+    }
+
+    let response;
+    try {
+      response = await request(endpoint, {
+        method: 'POST',
+        headers,
+        query: queryParams,
+        body: form,
+        dispatcher: proxyAgent,
+        signal: AbortSignal.timeout(this.connector.settings.timeout * 1000)
+      });
+    } catch (error) {
+      const message = this.getMessageFromError(error);
+
+      throw {
+        message: `Failed to reach file endpoint ${endpoint}; ${message}`,
+        retry: true
+      };
+    } finally {
+      if (!fileStream.closed) {
+        fileStream.close();
+      }
+    }
+
+    const ok = response.statusCode >= 200 && response.statusCode <= 299;
+    if (!ok) {
+      throw new Error(`HTTP request failed with status code ${response.statusCode} and message: ${await response.body.text()}`);
```

Comment:
Let's generate an OIBus error like in OIAnalytics.

```typescript
throw new OIBusError(
    `Error ${response.status}: ${response.statusText}`,
    [401, 403, 404, 500, 502, 503, 504].includes(response.status)
  );
```

This will trigger a retry if it's a network failure. It doesn't retry an error otherwise (if badly formatted query for example - status 400). 
Maybe we should also include auth error to keep retrying the same data. Otherwise all the data will be added in error folder